### ISSUE TITLE
Return `previous_operations` for next entry arguments

### ISSUE THREAD

**Issue by adzialocha** [p2panda/aquadoggo#52]

**Comment by sandreae** [comment]
I'm still not sure about this, it seems to confuse 2 areas which have different concerns: bamboo logs vs. p2panda documents. When performing an update on a document, presumably, you are already looking at the state of the current document (the document view). In which case, _this_ is where knowledge of previous operations should come from (it's the `DocumentViewId` you are updating).

I think previous operations should not be attached to next entry args :+1:

**Comment by cafca** [comment]
Yes, I agree. Clients should publish updates using the document view id that is most recent to them locally. This way we can resolve conflicts that we would ignore if updates were written to the document view that is most recent to the node.

**Comment by sandreae** [comment]
I'm reviving this issue ;-p

I've been thinking about this flow again (after a great demo app session yesterday) and maybe I'm coming round to the idea. Correct me if I'm wrong, but i _think_ the issue of getting out of date view ids back from the node, or somehow local client state getting out of sync, only applies when we assume some caching layer exists? Or maybe when batch publishing operations. But in these cases I think you'd have to account for this "out-of-sync-ness" through some app specific mechanisms anyway.

I could imagine `NextEntryArgs` looking like this:

```
type NextEntryArguments {
  """
  log id to be used to forge the next entry
  """
  logId: LogId!

  """
  sequence number to be used to forge the next entry
  """
  seqNum: SeqNum!

  """
  optional backlink hash to be used to forge the next entry
  """
  backlink: EntryHash

  """
  optional skiplink hash to be used to forge the next entry
  """
  skiplink: EntryHash

    """
  optional current document view id
  """
  viewId: DocumentViewId

}
```

It's optional as I can see two cases where it could be omitted:

1. These are the next entry args for a CREATE operation
2. The viewId contains only one hash and it's the backlink

**Comment by adzialocha** [comment]
I agree!

> Clients should publish updates using the document view id that is most recent to them locally.

We would need to define what we mean with "locally", indeed it can be a caching-layer but I would argue that the perspective of a client is always _through_ the node (and it can be a "locally" running node, but also a shared one). Potentially there is a caching layer involved as well, but hopefully it would be always as-close-as-possible to the node state.

**Comment by adzialocha** [comment]
> The viewId contains only one hash and it's the backlink

I wouldn't omit it maybe in that case .. it requires to "be smart" about it and I like the convenience of only looking at that field as a client?
